--- a/session.c	2026-08-11 10:22:17.000000000 +0800
+++ b/session.c	2026-09-02 18:33:23.893828144 +0800
@@ -1036,7 +1036,6 @@
 			child_set_env(&env, &envsize, "KRB5CCNAME", cp);
 	}
 
-#ifdef _AIX
 	{
 		char *cp;
 
@@ -1045,7 +1044,6 @@
 		read_environment_file(&env, &envsize, "/etc/environment",
 		    options.permit_user_env_allowlist);
 	}
-#endif
 #ifdef KRB5
 	if (s->authctxt->krb5_ccname)
 		child_set_env(&env, &envsize, "KRB5CCNAME",
