--- a/sshconnect2.c	2026-08-11 10:22:17.000000000 +0800
+++ b/sshconnect2.c	2026-09-02 18:33:47.050850056 +0800
@@ -1533,9 +1533,7 @@
 	struct stat st;
 
 	if (stat(id->filename, &st) == -1) {
-		do_log2(id->userprovided ?
-		    SYSLOG_LEVEL_INFO : SYSLOG_LEVEL_DEBUG3,
-		    "no such identity: %s: %s", id->filename, strerror(errno));
+		debug3("no such identity: %s", id->filename);
 		return NULL;
 	}
 	snprintf(prompt, sizeof prompt,
