--- a/src/enc/h264/v4l2m2m-impl.c
+++ b/src/enc/h264/v4l2m2m-impl.c
@@ -605,7 +605,13 @@ static int alloc_src_buffers_mmap(struct h264_encoder_v4l2m2m* self)
 				nvnc_log(NVNC_LOG_ERROR,
 						"Failed to mmap source buffer plane: %m");
 				return -1;
 			}
-			buffer->mmap_stride[j] = buffer->planes[j].bytesperline;
+			struct v4l2_format fmt = {
+				.type = self->src_type,
+			};
+			if (ioctl(self->fd, VIDIOC_G_FMT, &fmt) == 0)
+				buffer->mmap_stride[j] = fmt.fmt.pix_mp.plane_fmt[j].bytesperline;
+			else
+				buffer->mmap_stride[j] = buffer->planes[j].length / self->height;
 		}
 	}
 
