--- a/meson.build
+++ b/meson.build
@@ -40,13 +40,7 @@
 gio_dep = dependency('gio-2.0', version: glib_min_version)
 gtk_dep = dependency('gtk4', version: gtk_min_version)
 fribidi_dep = dependency('fribidi')
-appstream_dep = dependency('appstream',
-  fallback : ['appstream', 'appstream_dep'],
-  default_options : [
-    'systemd=false', 'apidocs=false', 'install-docs=false',
-    'stemming=false', 'svg-support=false', 'gir=false',
-  ],
-)
+appstream_dep = declare_dependency()
 
 target_system = host_machine.system()
 if target_system == 'darwin'
