--- a/src/meson.build
+++ b/src/meson.build
@@ -300,12 +300,6 @@
 gio_dep = dependency('gio-2.0', version: glib_min_version)
 gtk_dep = dependency('gtk4', version: gtk_min_version)
-appstream_dep = dependency('appstream',
-  fallback : ['appstream', 'appstream_dep'],
-  default_options : [
-    'systemd=false', 'apidocs=false', 'install-docs=false',
-    'stemming=false', 'svg-support=false', 'gir=false',
-  ],
-)
 
 libadwaita_deps = [
   dependency('glib-2.0', version: glib_min_version),
@@ -313,7 +307,6 @@ libadwaita_deps = [
   gio_dep,
   gtk_dep,
-  appstream_dep,
   cc.find_library('m', required: false),
 ]
