--- a/src/adw-about-dialog.c
+++ b/src/adw-about-dialog.c
@@ -7,7 +7,6 @@
 
 #include "config.h"
 #include <glib/gi18n-lib.h>
-#include <appstream.h>
 
 #include "adw-about-dialog.h"
 
@@ -1089,6 +1088,8 @@
 
   g_assert (self->appdata_resource_path != NULL);
 
+  return;
+
   appdata_uri = g_strconcat ("resource://", self->appdata_resource_path, NULL);
   appdata_file = g_file_new_for_uri (appdata_uri);
