diff --git a/configure.ac b/configure.ac
index 956bb85..0245158 100644
--- a/configure.ac
+++ b/configure.ac
@@ -11,11 +11,7 @@
 # SPDX-License-Identifier: LGPL-2.1-or-later
 
 AC_PREREQ(2.57)
-AC_INIT([strace],
-	st_esyscmd_s([./build-aux/git-version-gen .tarball-version]),
-	[strace-devel@lists.strace.io],
-	[strace],
-	[https://strace.io])
+AC_INIT([strace],[7.2])
 m4_define([copyright_year], st_esyscmd_s([./build-aux/copyright-year-gen .year]))
 m4_define([strace_manpage_date],
 	  st_esyscmd_s([./build-aux/file-date-gen doc/strace.1.in]))
