diff --git a/src/meson.build b/src/meson.build
--- a/src/meson.build
+++ b/src/meson.build
@@ -66,5 +66,5 @@
 		variables: [
 			'datarootdir=' + join_paths('${prefix}', get_option('datadir')),
-			'pkgdatadir=' + join_paths('${pc_sysrootdir}${datarootdir}', meson.project_name()),
+			'pkgdatadir=' + join_paths('${datarootdir}', meson.project_name()),
 			'wayland_scanner=${bindir}/wayland-scanner'
 		],
